packages/components/app/styles/components/dropdown.scss

```
@@ -0,0 +1,410 @@
+// >>>>>>>>>> WARNING <<<<<<<<<<<
+//
+// Notice: in this component we're using directly the `Hds::Button` component
+// (and adding a specialized class for the "toggle-button" variant, see below)
+// If you need to change the styling of the `Button` component, remember that this will impact also
+// this component too.
+// If instead you need to change only the styling of the `toggle-button` component, you can do it here using
+// the specialized class declared below.
+// This is NOT a standard approach that we use in the HDS design system implementation, but it's been
+// the least worst option we could find to solve the problem of sharing the exact same style of the
+// `Button (primary)` with other components.
+
+//
+// DROPDOWN COMPONENT
+//
+// properties within each class are sorted alphabetically
+// notice: pseudo-classes for the states *must* follow the order link > visited > hover > focus > active
+// see https://github.com/hashicorp/design-system-components/issues/132
+//
+//
+
+@use '../mixins/focus-ring' as *;
+
+$hds-dropdown-toggle-base-height: 36px;
+$hds-dropdown-toggle-border-radius: 5px;
+$hds-dropdown-toggle-border-width: 1px;
+
+
+
+// TOGGLE/ICON
+.hds-dropdown-toggle-icon {
+  align-items: center;
+  background-color: transparent;
+  border: $hds-dropdown-toggle-border-width solid transparent; // We need this to be transparent for a11y
+  border-radius: $hds-dropdown-toggle-border-radius;
+  box-sizing: border-box; // TODO https://github.com/hashicorp/design-system-components/issues/46
+  color: var(--token-color-foreground-strong);
+  display: flex;
+  height: $hds-dropdown-toggle-base-height;
+  isolation: isolate;
+  justify-content: center;
+  outline-style: solid; // used to avoid double outline+focus-ring in Safari (see https://github.com/hashicorp/design-system-components/issues/161#issuecomment-1031548656)
+  outline-color: transparent; // We need this to be transparent for a11y
+  padding: 1px;
+  position: relative;
+  min-width: $hds-dropdown-toggle-base-height;
+
+  &:hover,
+  &.is-hover {
+    background-color: var(--token-color-surface-interactive);
+    border-color: var(--token-color-border-strong);
+    cursor: pointer;
+  }
+
+  &:focus,
+  &.is-focus {
+    @include hds-focus-ring-with-pseudo-element(
+      $top: -1px,
+      $right: -1px,
+      $bottom: -1px,
+      $left: -1px,
+      $radius: $hds-dropdown-toggle-border-radius
+    );
+  }
+
+  &:active,
+  &.is-active {
+    background-color: var(--token-color-surface-interactive-active);
+    border-color: var(--token-color-border-strong);
+  }
+}
+
+.hds-dropdown-toggle-icon__wrapper {
+  align-items: center;
+  border-radius: 3px; // $hds-dropdown-toggle-border-radius- 1px padding - 1px border
+  display: flex;
+  justify-content: center;
+  height: 2rem;
+  padding: 1px;
+  width: 2rem;
+
+  img {
+    border-radius: inherit;
```

if we add an `object-fit: cover` we are resilient to the possibility that the image may not be square.

```suggestion
    border-radius: inherit;
    object-fit: cover;
```